app/routes.py

```diff
@@ -1,2 +1,325 @@
+                "title": video.title,
+                "release_date": video.release_date,
+                "total_inventory": video.total_inventory,
+                "available_inventory": available
+        }
+
+        video_response.append(response_body)
+
+    #video_response = [video.make_json() for video in videos]
+
+    return make_response(jsonify(video_response),200)
+
+
+@video_bp.route("/<video_id>", methods=["GET"], strict_slashes=False)
+def get_video_by_id(video_id):
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return make_response({"details": f"Video with id : {video_id} not found"}, 404)
+    
+    currently_rented_out = len(video.rentals)
+    total = video.total_inventory
+    available = total - currently_rented_out
+
+    response_body = {
+            "id": video.id,
+            "title": video.title,
+            "release_date": video.release_date,
+            "total_inventory": video.total_inventory,
+            "available_inventory": available
+    }
+
+    return make_response(response_body, 200)
+
+
+@video_bp.route("/<video_id>", methods=["PUT"], strict_slashes=False)
+def update_video_by_id(video_id):
+    video = Video.query.get(video_id)
+    request_data = request.get_json()
+    
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+    
+    if not request_data.get("title") or type(request_data.get("title")) != str:
+        return make_response({"details": "Please enter a valid value for title field."}, 400)
+    elif not request_data.get("release_date") or type(request_data.get("release_date")) != str:
+        return make_response({"details": "Please enter a valid value for release_date field."}, 400)
+    elif not request_data.get("total_inventory")  or type(request_data.get("total_inventory")) != int:
+        return make_response({"details": "Please enter a valid value for total_inventory field."}, 400)
+    
+    request_data = request.get_json()
+
+    video.title = request_data["title"]
+    video.release_date = request_data["release_date"]
+    video.total_inventory = request_data["total_inventory"]
+
+    db.session.commit()
+
+    currently_rented_out = len(video.rentals)
+    total = video.total_inventory
+    available = total - currently_rented_out
+
+    response_body = {
+            "id": video.id,
+            "title": video.title,
+            "release_date": video.release_date,
+            "total_inventory": video.total_inventory,
+            "available_inventory": available
+    }
+
+    return make_response(response_body, 200)
+
+@video_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get(video_id)
+
+    if video is None:
+        return make_response({"details": f"Video id number: {video_id} not found"}, 404)
+
+    db.session.delete(video)
+    db.session.commit()
+
+    return make_response({"id": f"{video.id}"}, 200)
```

Comment:
Again an integer is best here.

```suggestion
    return make_response({"id": video.id }, 200)
```